woocommerce/woocommerce-android
# Domain change: Remove feature flag

Fixes #8467, a subtask of #8238.

This PR removes the feature flag for the domain change and updates the release notes. 

I also added 2 minor fixes -- a preselection fix for the free domain search and a domain price cache.

**To test:**
1. Try the free domain search and make sure a domain is correctly selected when you don't change the pre-selected item
2. Try the paid domain search and verify the domain selection does not log anymore
